tests/Cache/CacheRateLimiterTest.php

```diff
@@ -39,13 +39,22 @@ public function testTooManyAttemptsReturnsTrueIfMaxAttemptsExceeded()
     public function testHitProperlyIncrementsAttemptCount()
     {
         $cache = m::mock(Cache::class);
-        $cache->shouldReceive('add')->once()->with('key', 1, 1);
+        $cache->shouldReceive('add')->once()->with('key', 0, 1);
         $cache->shouldReceive('increment')->once()->with('key');
         $rateLimiter = new RateLimiter($cache);
 
         $rateLimiter->hit('key', 1);
     }
 
+    public function testRetriesLeftReturnsCorrectCount()
```

Comment:
The only mocked part is the caching of current attempts (set to 3).

The logic of `retiresLeft()` is not mocked. If you run that test now on the current unmodified code - it fails and returns `3`.

It should be `2` - which this test detects.